File: MultiLanguageSandbox/src/test/deps/lua/7.lua

--[[
Calculates the area of a triangle given its base and height.

Parameters:
- base (number): The base length of the triangle.
- height (number): The height of the triangle.

Returns:
- number: The calculated area of the triangle, rounded to one decimal place.

Examples:
local area = calculate_triangle_area(1, 2)
print(area)  -- Output: 1.0

local area = calculate_triangle_area(3, 4)
print(area)  -- Output: 6.0
--]]

function calculate_triangle_area(base, height)
    local area = 0.5 * base * height
    return math.floor(area * 10 + 0.5) / 10
end
local function test_calculate_triangle_area()
    assert(math.abs(calculate_triangle_area(1, 2) - 1.0) < 1e-6)
    assert(math.abs(calculate_triangle_area(3, 4) - 6.0) < 1e-6)
    assert(math.abs(calculate_triangle_area(5, 8) - 20.0) < 1e-6)
    assert(math.abs(calculate_triangle_area(7, 3) - 10.5) < 1e-6)
    assert(math.abs(calculate_triangle_area(10, 10) - 50.0) < 1e-6)

    -- print("All tests passed.")
end

test_calculate_triangle_area()

File: MultiLanguageSandbox/src/test/deps/typescript/7.ts

/**
 * Calculate the area of a triangle given its base and height.
 * 
 * @param base The base length of the triangle.
 * @param height The height of the triangle.
 * 
 * @returns The calculated area of the triangle, rounded to one decimal place.
 * 
 * Examples:
 * calculateTriangleArea(1, 2) returns 1.0
 * calculateTriangleArea(3, 4) returns 6.0
 */

function calculateTriangleArea(base: number, height: number): number {
    const area = (base * height) / 2;
    return Math.round(area * 10) / 10;
}
const testCalculateTriangleArea = () => {
    // The equivalent of C's assertion with a floating-point comparison in TypeScript.
    const assertCloseTo = (value: number, expected: number, tolerance: number = 1e-6) => {
        console.assert(Math.abs(value - expected) < tolerance);
    };

    assertCloseTo(calculateTriangleArea(1, 2), 1.0);
    assertCloseTo(calculateTriangleArea(3, 4), 6.0);
    assertCloseTo(calculateTriangleArea(5, 8), 20.0);
    assertCloseTo(calculateTriangleArea(7, 3), 10.5);
    assertCloseTo(calculateTriangleArea(10, 10), 50.0);

    // Uncomment to log the outcome to the console.
    // console.log("All tests passed.");
};

testCalculateTriangleArea();